packages/web-components/src/components/va-alert-expandable/va-alert-expandable.tsx

```diff
@@ -0,0 +1,150 @@
+  @Element() el: HTMLElement;
+
+  @State() open: boolean;
+
+  /**
+   * Determines the icon and background color.
+   * One of `info`, `error`, `success`, `warning`, or `continue`
+   */
+  @Prop() status: string = 'info';
+
+  /**
+   * The text to trigger the expansion
+   */
+  @Prop() trigger: string = '';
+
+  /**
+   * If true, doesn't fire the CustomEvent which can be used for analytics tracking.
+   */
+  @Prop() disableAnalytics: boolean = false;
+
+  /**
+   * If false, the status icon is removed.
+   */
+  @Prop() showIcon: boolean = true;
+
+  /**
+   * If true, the alert will be visible.
+   */
+   @Prop() visible: boolean = true;
+
+  /**
+   * Fires when the component has successfully finished rendering for the first
+   * time.
+   */
+   @Event({
+    eventName: 'va-component-did-load',
+    composed: true,
+    bubbles: true,
+  })
+  vaComponentDidLoad: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  @Listen('resize', { target: 'window' })
+  handleResize() {
+    this.updateAlertBodyMaxHeight();
+  }
+
+  toggleOpen(): void {
+    // Conditionally track the event.
+    if (!this.disableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-alert-expandable',
+        action: !this.open ? 'expand' : 'collapse',
+        details: {
+          triggerText: this.trigger,
+        },
+      });
+    }
+
+    this.open = !this.open;
+  }
+
+  handleKeydown(event): void {
+    if (event.key === ' ' || event.key === 'Enter') {
+      event.preventDefault();
+      this.toggleOpen();
+    }
+  }
+
+  // Ensures that the CSS animation is consistent and uses the correct max-height for its content
+  updateAlertBodyMaxHeight() {
+    const bodyElm = this.el.shadowRoot.getElementById('alert-body');
+    const contentHeight = bodyElm.scrollHeight + 'px';
+    // the additional 2em is #alert-body margin-top and margin-bottom when open
+    bodyElm.style.setProperty(
+      '--calc-max-height',
+      'calc(' + contentHeight + ' + 2rem)',
+    );
+  }
+
+  componentDidLoad() {
+    this.updateAlertBodyMaxHeight();
+    this.vaComponentDidLoad.emit();
+  }
+
+  render() {
+    const { status, open, showIcon, visible } = this;
+    const alertClasses = classnames('alert-expandable', status, { 'hide-icon': !showIcon });
+    const bodyClasses = classnames('alert-expandable-body', { 'open': open, 'closed': !open });
+    const role = status === 'error' ? 'alert' : null;
+    const ariaLive = status === 'error' ? 'assertive' : null;
+
+    if (!visible) return <div aria-live="polite" />;
+
+    return (
+      <Host>
+        <div role={this.el.getAttribute('data-role') || role} aria-live={ariaLive} class={alertClasses}>
```

Comment:
Do we need to set the role with `data-role`? If someone were to do this:

```html
<va-alert-expandable role="something"></va-alert-expandable>
```

That `role` attribute would essentially end up on the `<Host>`

So we might be able to do something like this to override it when we have an error:

```tsx

<Host role={status === 'error' ? 'alert' : this.el.getAttribute('role')}>

Reply:
I'll let you all guide this. This setup is currently how `va-alert` currently operates. I would assume whatever is done on one component should be done on the other. If we did move the role up to the host, we would need to move the aria-live attribute, as well.

Reply:
@bkjohnson yes this was a specific use case due to banner vs alert.
Here's the details, for all Alerts we have an aria-role of alert however with the Banner component we wanted to change that to aria-role of banner since it is a better indicator of what the component actually is from an accessibility perspective. Since we nest a va-alert inside of va-banner I needed a way to pass down the correct role depending on the component we were using. 
Moral of all this is we don't need this.el.getAttribute('data-role') here instead just set the aria-role we want to use here ie: role={role}